src/App.tsx

```diff
@@ -1,61 +1,38 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { TodosWithUsers } from './type.ts/TodosWithUser';
+import { TodoForm } from './components/TodoForm';
+import { getUserById } from './utils/userId';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+
+const todos: TodosWithUsers[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId) || null,
+}));
 
 export const App = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <form action="/api/todos" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>
-              Choose a user
-            </option>
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
+  const [todosArr, setTodosArr] = useState(todos);
 
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
+  const addTodo = (newTodo: TodosWithUsers) => {
+    const maxId =
+      todosArr.length > 0 ? Math.max(...todosArr.map(todo => todo.id)) : 0;
```

Comment:
The `maxId` calculation should use `todosArr` instead of `todos` to ensure correct ID assignment. This will prevent potential issues when adding new todos.